src/jdk.compiler/share/classes/com/sun/tools/javac/code/Lint.java

```diff
@@ -385,6 +391,15 @@ public boolean isSuppressed(LintCategory lc) {
         return suppressedValues.contains(lc);
     }
 
+    /**
+     * Helper method. Log a lint warning if its lint category is enabled.
+     */
+    public void logIfEnabled(DiagnosticPosition pos, LintWarning warning) {
+        if (isEnabled(warning.getLintCategory())) {
```

Comment:
I'm not 100% sure about this method. On the one hand it makes clients simpler - but it does require `Lint` keeping track of a `log`. An alternative could be to add a method in `AbstractLog`, e.g.:
```
warningIfEnabled(DiagnosticPosition pos, Lint lint, LintWarning key)
```

Reply:
elsewhere, @archiecobbs suggested to maybe make `log` a parameter of the method. I like the simplicity of the suggestion, and I will go with it.